app/src/main/java/org/odk/odknotifications/Activities/MainActivity.java

```diff
@@ -305,6 +292,20 @@ private void addMenuItemInNavMenuDrawer() {
         navView.invalidate();
     }
 
+    private void setUserName(){
+
+        loggedInUsername = getActiveUser();
+        Log.e("Success", "Database available " + loggedInUsername);
+        if(loggedInUsername!=null){
+            if(!(loggedInUsername.compareTo("anonymous")==0)&& loggedInUsername.length()>8 && loggedInUsername.substring(0,9).compareTo("username:")==0){
```

Comment:
Please use `equals` or `equalsIgnoreCase` when you're doing string comparison you can simplify it a little bit like `("anonymous").equals(loggedInUsername)` this way you will not have to do the null check. And also please mention why're you doing substring operation over here. I mean a comment would be sufficient explaining that this is the expected string in loggedInUsername and when we do substring it should return me something like this.

Reply:
Done. Also, this code block was a part of the original code I just moved it to a separate method. So if there is any specific reason for this implementation @DreamyPhobic may comment.